### PR TITLE
authenticate via facebook

### PR DESCRIPTION
**Pull request by ninhunest**

### CODE REVIEW

**Comment by ninhunest** [comment]
ready

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
-  devise_for :users
   root "static_pages#home"
+  devise_for :users,
+   :controllers => {:omniauth_callbacks => "users/omniauth_callbacks" }
```
Comment (dothidiemthao) [inline config/routes.rb]: controller: {omniauth_callbacks: "users/omniauth_callbacks"}

File: app/models/user.rb [hunk 2]
```diff
@@ -9,8 +9,24 @@ class User < ApplicationRecord
   has_many :following, through: :active_relationships, source: :followed
   has_many :followers, through: :passive_relationships, source: :follower
   devise :database_authenticatable, :registerable, :recoverable,
-   :rememberable, :trackable, :validatable
-
+    :rememberable, :trackable, :validatable, :omniauthable,
+    :omniauth_providers => [:facebook]
```
Comment (dothidiemthao) [inline app/models/user.rb]: omniauth_providers: :facebook

**Comment by dothidiemthao** [comment]
commented

**Comment by ninhunest** [comment]
ready

**Pull request merged**